frontend/src/lib-common/utils/absences-date-range.spec.ts

```diff
@@ -0,0 +1,112 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { uniqueId } from 'lodash'
+
+import { Absence } from 'lib-common/api-types/child/Absences'
+import LocalDate from 'lib-common/local-date'
+import { groupAbsencesByDateRange } from './absences'
+
+describe('absences date range', () => {
+  describe('grouping works with', () => {
+    it('a single date', () => {
+      const absence: Absence = {
+        id: uniqueId(),
+        childId: uniqueId(),
+        date: LocalDate.today().addDays(1),
+        absenceType: 'SICKLEAVE',
+        careType: 'PRESCHOOL'
+      }
+      const absences = [absence]
+      expect(groupAbsencesByDateRange(absences)[0].durationInDays()).toBe(-1)
+    })
+
+    it('two dates', () => {
+      const absence: Absence = {
+        id: uniqueId(),
+        childId: uniqueId(),
+        date: LocalDate.today().addDays(1),
+        absenceType: 'SICKLEAVE',
+        careType: 'PRESCHOOL'
+      }
+      const absence2: Absence = {
+        id: uniqueId(),
+        childId: uniqueId(),
+        date: LocalDate.today().addDays(2),
+        absenceType: 'SICKLEAVE',
+        careType: 'PRESCHOOL'
+      }
+      const absences = [absence, absence2]
+      expect(groupAbsencesByDateRange(absences)[0].durationInDays()).toBe(-2)
```

Comment:
why are these negative?

Reply:
FiniteDateRange.durationInDays() returns the duration as negative values. Maybe it would return positive numbers if instead of start.differenceInDays(end) we would use end.differenceInDays(start).

Reply:
lets change it that way